src/main/java/com/openlattice/datastore/directory/controllers/PrincipalDirectoryController.java

```diff
@@ -235,4 +228,31 @@ public String getDbAccessCredential() {
     public AuthorizationManager getAuthorizationManager() {
         return authorizations;
     }
+
+
+    @Override
+    @PostMapping(
+            path = UPDATE,
+            consumes = MediaType.APPLICATION_JSON_VALUE )
+    public Void addPrincipalToPrincipal( @RequestBody DirectedAclKeys directedAclKeys ) {
+        ensureWriteAccess( directedAclKeys.getTarget() );
+        ensureWriteAccess( directedAclKeys.getSource() );
```

Comment:
Shouldn't you have owner on source?

Reply:
you need to own a principal to grant it to someone?

Reply:
I guess that makes sense if it's a permissions grant